scalafmt-tests/src/test/scala/org/scalafmt/config/ConfigDialectOverrideTest.scala

```diff
@@ -0,0 +1,80 @@
+package org.scalafmt.config
+
+import scala.meta.Dialect
+
+import munit.FunSuite
+
+class ConfigDialectOverrideTest extends FunSuite {
+
+  // toplevelSeparator is never actually used,
+  // but as the only non-boolean Dialect value it makes for a good test
+  test("dialect override - non boolean setting") {
+    ScalafmtConfig.fromHoconString(
+      """|
+         |runner.dialectOverride.toplevelSeparator = ">"
+         |runner.dialect = scala213
+         |""".stripMargin,
+    ).get
+  }
+
+  test("throws on an incorrect type of setting") {
+    intercept[java.util.NoSuchElementException] {
+      ScalafmtConfig.fromHoconString(
+        """|
+           |runner.dialectOverride.toplevelSeparator = true
+           |runner.dialect = scala213
+           |""".stripMargin,
+      ).get
+    }
+  }
+
+  def testBooleanFlag(methodName: String, getter: Dialect => Boolean): Unit = {
+    def makeBooleanConfig(setting: String, value: Boolean) = ScalafmtConfig
+      .fromHoconString(
+        s"""|
+            |runner.dialectOverride.$setting = $value
+            |runner.dialect = scala213
+            |""".stripMargin,
+      ).get
+    Seq(true, false).foreach { flag =>
+      test(s"boolean flag: $methodName($flag)") {
+        assertEquals(
+          getter(makeBooleanConfig(methodName, flag).runner.getDialect),
+          flag,
+        )
+      }
+    }
+  }
+
+  testBooleanFlag("allowFewerBraces", _.allowFewerBraces)
+  testBooleanFlag("withAllowFewerBraces", _.allowFewerBraces)
+  testBooleanFlag("useInfixTypePrecedence", _.useInfixTypePrecedence)
+  testBooleanFlag("withUseInfixTypePrecedence", _.useInfixTypePrecedence)
+  testBooleanFlag(
+    "allowImplicitByNameParameters",
+    _.allowImplicitByNameParameters,
+  )
+  testBooleanFlag(
+    "withAllowImplicitByNameParameters",
+    _.allowImplicitByNameParameters,
+  )
+  testBooleanFlag("allowSignificantIndentation", _.allowSignificantIndentation)
+  testBooleanFlag(
+    "withAllowSignificantIndentation",
+    _.allowSignificantIndentation,
+  )
+
+  test("applying generated boolean map elements does not result in errors") {
```

Comment:
it's a very good negative test. i am proposing, above, to add some positive asserts as well.